basics/list-comprehension.md

```diff
@@ -0,0 +1,77 @@
+# List Comprehension
+
+
+## Definition
+List comprehensions creates a new list from an existing list that is a shorter syntax than normal. It also processes the list much faster than using a for loop.
+
+
+## Without List Comprehension
+```python
+animals = ["dog", "cat", "giraffe", "donkey", "ape"]
+arr = []
+
+for i in animals:
+  if len(i) == 3:
+    arr.append(i)
+    
+print(arr)
+```
+```python
+['dog', 'cat', 'ape']
+```
+The example above shows a list of animals and the for loop iterates through the list of animals and the if statement filters the list and only adds the animal if their name is exactly 3 characters long.
+
+## With List Comprehension
+```python
+animals = ["dog", "cat", "giraffe", "donkey", "ape"]
+
+arr = [i for i in animals if len(i) == 3]
+print(arr)
+```
+```python
+['dog', 'cat', 'ape']
+```
+The example above does exactly the same thing but is in a much more concise syntax where it iterates through the list of animals and appends to the new list if their name is exactly 3 characters long.
+
+## Syntax
```

Comment:
What's the purpose of this section? Reading a description syntax isn't a good way to learn, in my experience. It would be better to just show the `i.lower()` example, without trying to describe the syntax with words too much.